pageserver/src/tenant/remote_timeline_client/index.rs

```diff
@@ -233,6 +251,9 @@ pub struct LayerFileMetadata {
     #[serde(default = "ShardIndex::unsharded")]
     #[serde(skip_serializing_if = "ShardIndex::is_unsharded")]
     pub shard: ShardIndex,
+
+    #[serde(skip_serializing_if = "Option::is_none", default)]
+    pub encrypted_with_key_version: Option<u32>,
```

Comment:
"`encrypted_with_key_version`" is a lot of bytes (we repeat it a lot of times), it would be nicer to have a shorter name, like `encryption_key`.

Reply:
It's not actually `key`, so `encryption_key_version` would be better in my opinion.
Taking into account that in `IndexPart` it's just `keys` field, I think `key_version` here would be fine too

Reply:
I renamed it to `encryption_key`